### PR TITLE
http://aqicn.org

### PR DESCRIPTION
**Pull request by beetleman**
Air quality plugin. Its use api from http://aqicn.org.

### CODE REVIEW

**Comment by lasers** [comment]
Hi. I'm a tester for your air quality plugin.
This does not work for me out of the box using default options right now.

Using Python2.
```
lasers~/src/py3status/py3status/modules git:(beetleman-aqicn) python2 aqicn.py 
Traceback (most recent call last):
  File "aqicn.py", line 68, in <module>
    from utllib2 import urlencode
ImportError: No module named utllib2
```
Using Python3.
```
lasers~/src/py3status/py3status/modules git:(beetleman-aqicn) python3 aqicn.py
Traceback (most recent call last):
  File "aqicn.py", line 175, in <module>
    module_test(Py3status)
  File "/home/chris/src/py3status/py3status/module_test.py", line 52, in module_test
    print(getattr(module, method)(*method_args))
  File "aqicn.py", line 166, in aqicn
    'color': self._color(data)
  File "aqicn.py", line 157, in _color
    self, color_key_default, self.color_unknown
AttributeError: 'Py3status' object has no attribute 'color_unknown'
```

File: py3status/modules/aqicn.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+"""
+
+try:
+    # python 3
+    from urllib.error import URLError
+    from urllib.request import urlopen, Request
+    from urllib.parse import urlencode
+except ImportError:
+    # python 2
+    from utllib2 import urlencode
```
Comment (pydsigner) [inline py3status/modules/aqicn.py]: Should be `urllib2`.

**Comment by beetleman** [comment]
sorry I forgot test It on py27 and without my configuration. Im so embarrassed. It should work now

File: py3status/modules/aqicn.py (hunk 1 above)
Comment (beetleman) [inline py3status/modules/aqicn.py]: `from urllib import urlencode` :smile_cat:. Python2 is Messy.

**Comment by lasers** [comment]
Worked okay out of the box. 👍 Can you check for typos too?

Add missing `Requires` section for `requests`... (See example from `github.py`)
```
Requires:
    python-requests: Python HTTP for Humans
```

Remove `color_unknown ` in option/example too if you know it is not going to be used too (ie default color). I don't know if it's being used...? Same thing with `{pm25}`, but I think that one may be coming in from json. (I noticed pm25 is the only occurrence word there). 

Where did you get this information from?
`best option is choice uid instead city name: city = '@8691'`

If there is bad information in `city`, it could lead to an exception easily. Just mentioning.

`{aqi}` and `{pm25}` both gave me same number. May be normal / expected.
It seems to be working okay for me here.

The nearest available air quality station (on the website) gave me `Good` status.
Thank you for this module.

**Comment by beetleman** [comment]
This module dont have any requires (i dont want force anybody to install extra modules)
I dont have time today for it but i ill add more tomorrow. {pm25} is parametr from API, every station can have different set of params.
TODO
- [ ] doc
- [ ] geolocation

**Comment by tobes** [comment]
@beetleman Thanks for this I like the idea of the module :)  I'm only seeing Shanghai data but that might be because I have no api token (some info on how to get one would be great).

I'd allow more of the data to be available to the user some via documentation eg `{o3}`  and some like `{city}` that I'd add.  Also how do we set a specific station in cities with many

Some work will be needed on the naming of the config parameters, but should be fairly simple.  I'd really like the colors etc to come via thresholds but that might not be possible as you also get descriptions at the same time.

> This module dont have any requires

cool, this is good.

**Comment by tobes** [comment]
I'd also be inclined to call the module something more clear like `air_quality`

**Comment by ultrabug** [comment]
@beetleman thanks again for your contribution. How do you feel about the remarks here ? I'd love your code to be merged :)

**Comment by beetleman** [comment]
+1 for change name. I love to have more time to finish it;(. If its ok in this form You can merge it and I'll  improve it over time because i using it:D

**Comment by lasers** [comment]
No problem. I can take on this and make changes from here. Cheers.

**Comment by ultrabug** [comment]
Okay @beetleman I'll merge when the CI tests pass, please correct the following

> `cache_timeout` (default 3600) does not match docstring 900

Then we can iterate from there, thanks again !

**Comment by tobes** [comment]
>  I'll merge when the CI tests pass, please correct the following

I think it needs more work before any merge.  @lasers has been messing with this PR in #800 

I really like the idea of this module

**Comment by ultrabug** [comment]
Well if @beetleman is okay, we maybe should merge this and work from here on #800 

This would allow us to move forward without having to bug @beetleman all the time

**Comment by tobes** [comment]
@beetleman the updated version of your PR #800 is now finished and imho ready for merging should you wish to review it.

Thanks for this fun new module

**Comment by beetleman** [comment]
Thanks guys for make my little hacks looks like real module. Sorry for leaving you all with my code but i have little time for side projects this days;(

**Comment by ultrabug** [comment]
Merged as per #800 !

**Pull request closed without merging**